### ISSUE TITLE
index.php not loading (blank screen) no error messages

### ISSUE THREAD

**Issue by richie2000-1** [rvolz/BicBucStriim#399]
Hello,
recently the page (index.php) no longer loads. The installcheck.php page works and doesn't show me any (red) errors either.
I have already put the index.php in debug mode and also received a log file. but this has only two entries, which do not indicate an error for me.
INFO - 2023-07-18T20:46:15+08:00 - BicBucStriim 1.5.3: Running in debug mode.
INFO - 2023-07-18T20:46:15+08:00 - Running on PHP: 8.2.0
I have bbs running on a QNAP TS431P3.
Firmware version QTS 5.1.0.2444 Build 20230629

**Comment by ReWo77** [comment]
Hello,
I have same problem on my QNAP TS-251A. After upgrading to QTS 5.1.0.2444 I get a blank screen.
Same output from the installcheck.php (no red errors) and the debug mode:
INFO - 2023-07-20T01:46:11+08:00 - BicBucStriim 1.5.3: Running in debug mode.
INFO - 2023-07-20T01:46:11+08:00 - Running on PHP: 8.2.0

**Comment by rvolz** [comment]
It might be the same problem as mentioned [here](https://github.com/rvolz/BicBucStriim/issues/398#issuecomment-1636844772)

**Comment by richie2000-1** [comment]
I downloaded version 1.6.4. Unfortunately with the same result...

**Comment by ReWo77** [comment]
Thanks, it worked for me with the upgrade to 1.6.4

**Comment by richie2000-1** [comment]
with PHP-Error-output I got:

`Deprecated: Return type of Slim\Helper\Set::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Helper/Set.php on line 178

Deprecated: Return type of Slim\Helper\Set::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Helper/Set.php on line 183

Deprecated: Return type of Slim\Helper\Set::offsetSet($offset, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Helper/Set.php on line 188

Deprecated: Return type of Slim\Helper\Set::offsetUnset($offset) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Helper/Set.php on line 193

Deprecated: Return type of Slim\Helper\Set::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Helper/Set.php on line 202

Deprecated: Return type of Slim\Helper\Set::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Helper/Set.php on line 211

Deprecated: Return type of Slim\Middleware\Flash::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Middleware/Flash.php on line 158

Deprecated: Return type of Slim\Middleware\Flash::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Middleware/Flash.php on line 168

Deprecated: Return type of Slim\Middleware\Flash::offsetSet($offset, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Middleware/Flash.php on line 178

Deprecated: Return type of Slim\Middleware\Flash::offsetUnset($offset) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Middleware/Flash.php on line 186

Deprecated: Return type of Slim\Middleware\Flash::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Middleware/Flash.php on line 195

Deprecated: Return type of Slim\Middleware\Flash::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Middleware/Flash.php on line 205

Deprecated: Return type of Slim\Environment::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Environment.php on line 186

Deprecated: Return type of Slim\Environment::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Environment.php on line 194

Deprecated: Return type of Slim\Environment::offsetSet($offset, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Environment.php on line 206

Deprecated: Return type of Slim\Environment::offsetUnset($offset) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Environment.php on line 214

Deprecated: Return type of Slim\Environment::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Environment.php on line 224

Deprecated: Creation of dynamic property RedBean_QueryWriter_SQLiteT::$sqltype_typeno is deprecated in /share/MD0_DATA/Web/bbs/vendor/rb.php on line 2576

Deprecated: Creation of dynamic property RedBean_AssociationManager::$toolbox is deprecated in /share/MD0_DATA/Web/bbs/vendor/rb.php on line 4866

Deprecated: Creation of dynamic property RedBean_AssociationManager_ExtAssociationManager::$toolbox is deprecated in /share/MD0_DATA/Web/bbs/vendor/rb.php on line 4866

Fatal error: During inheritance of ArrayAccess: Uncaught ErrorException: Return type of Slim\Http\Response::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Http/Response.php:451 Stack trace: #0 /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Http/Response.php(46): Slim\Slim::handleErrors(8192, 'Return type of ...', '/share/MD0_DATA...', 451) #1 /share/MD0_DATA/Web/bbs/vendor/composer/ClassLoader.php(582): include('/share/MD0_DATA...') #2 /share/MD0_DATA/Web/bbs/vendor/composer/ClassLoader.php(433): Composer\Autoload\{closure}('/share/MD0_DATA...') #3 /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Slim.php(167): Composer\Autoload\ClassLoader->loadClass('Slim\\Http\\Respo...') #4 /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Helper/Set.php(228): Slim\Slim->Slim\{closure}(Object(Slim\Helper\Set)) #5 /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Helper/Set.php(88): Slim\Helper\Set->Slim\Helper\{closure}(Object(Slim\Helper\Set)) #6 /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Slim.php(232): Slim\Helper\Set->get('response') #7 /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Slim.php(1064): Slim\Slim->__get('response') #8 /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Middleware/PrettyExceptions.php(73): Slim\Slim->contentType('text/html') #9 /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Slim.php(1300): Slim\Middleware\PrettyExceptions->call() #10 /share/MD0_DATA/Web/bbs/index.php(218): Slim\Slim->run() #11 {main} in /share/MD0_DATA/Web/bbs/vendor/slim/slim/Slim/Http/Response.php on line 46`

**Comment by ReWo77** [comment]
I followed the instructions for upgrading (https://github.com/mikespub-org/rvolz-BicBucStriim).
Then I changed something in the index.php from "development" to "production" and it works...

**Comment by richie2000-1** [comment]
> I followed the instructions for upgrading (https://github.com/mikespub-org/rvolz-BicBucStriim). Then I changed something in the index.php from "development" to "production" and it works...

thanks, now it works...

**Comment by mikespub** [comment]
Sorry about that, and thanks to @ReWo77 for a quick fix :-)

In php.ini:
```
# Running slim/slim 2.x on PHP 8.2 needs php error_reporting set to E_ALL & ~E_DEPRECATED & ~E_STRICT (= production default)
error_reporting=E_ALL & ~E_DEPRECATED & ~E_STRICT
```

This can be done either in the php.ini file as provided in the source code, or configured in index.php as mentioned above.

Both are included in the new release 1.6.5 at https://github.com/mikespub-org/rvolz-BicBucStriim
